core/templates/dev/head/pages/exploration_editor/ExplorationStatesService.js

```diff
@@ -256,35 +259,36 @@ oppia.factory('ExplorationStatesService', [
       },
       saveConfirmedUnclassifiedAnswers: function(stateName, newAnswers) {
         saveStateProperty(
-          stateName, 'confirmed_unclassified_answers', newAnswers);
+          stateName, 'confirmed_unclassified_answers', newAnswers, true);
       },
       getInteractionDefaultOutcomeMemento: function(stateName) {
         return getStatePropertyMemento(stateName, 'default_outcome');
       },
       saveInteractionDefaultOutcome: function(stateName, newDefaultOutcome) {
-        saveStateProperty(stateName, 'default_outcome', newDefaultOutcome);
+        saveStateProperty(
+          stateName, 'default_outcome', newDefaultOutcome, true);
       },
       getHintsMemento: function(stateName) {
         return getStatePropertyMemento(stateName, 'hints');
       },
       saveHints: function(stateName, newHints) {
-        saveStateProperty(stateName, 'hints', newHints);
+        saveStateProperty(stateName, 'hints', newHints, true);
       },
       getSolutionMemento: function(stateName) {
         return getStatePropertyMemento(stateName, 'solution');
       },
       saveSolution: function(stateName, newSolution) {
-        saveStateProperty(stateName, 'solution', newSolution);
+        saveStateProperty(stateName, 'solution', newSolution, true);
       },
       getContentIdsToAudioTranslationsMemento: function(stateName) {
         return getStatePropertyMemento(
           stateName, 'content_ids_to_audio_translations');
       },
       saveContentIdsToAudioTranslations: function(
-          stateName, newContentIdsToAudioTranslations) {
+          stateName, newContentIdsToAudioTranslations, changeListRequired) {
```

Comment:
I agree! have made the changes accordingly. Also, I'm thinking to introduce new commands for translations (instead of edit_state_property we will use translate_state_property which will have content_id language nevValue and oldValue as edit_state_property use to send the entire content_ids_to_audio_translation dict to the server.)